### ISSUE TITLE
Allow pre-set hostnames to be used during Elemental registration

### ISSUE THREAD

**Issue by rdoxenham** [rancher/elemental-operator#591]
At present, when a system registers via the Elemental operator it uses a unique ID for the system hostname. It would be great if we could tell Elemental to consume the existing hostname of the system, e.g. because it was defined by DHCP (or some other means). For some use-cases it's not desirable to have a random system hostname set.

One example here would be to have the configuration as follows:

```
config:
  cloud-config:
    users:
      - name: root
        passwd: root
  elemental:
    install:
      device: /dev/nvme0n1
      poweroff: false
    registration:
      auth: tpm
    reset:
      reboot: true
      reset-oem: true
      reset-persistent: true
machineName: ${System Information/Hostname}
```

Thank you!

**Comment by frelon** [comment]
The new key is `${System Data/Runtime/Hostname}`, I will test this today and also add it to the documentation :+1:

**Comment by fgiudici** [comment]
Tested #593 successfully, good work @frelon !
The host current hostname is used as the MachineInventory name created on the operator side and is later enforced as static hostname when doing the k3s/RKE2 provisioning.

Still if booting via the SLE micro for Rancher ISO images shipped through Elemental, we set there a static hostname "Rancher-XXXXX" and that is what would be picked if one uses the `machineName: ${System Data/Runtime/Hostname}` config snippet.
Anyway, this should be managed changing how static hostname is set in Elemental SLE Micro for Rancher base ISOs, and will be tracked apart.

Closing, this will land in 1.5.x elemental-register releases.
